### ISSUE TITLE
Improve naming and documentation of `noise_scaling_converter`

### ISSUE THREAD

**Issue by Misty-W** [unitaryfund/mitiq#1244]
Pre-Request Checklist
---------------------

- [X] I checked to make sure that this feature has not already been requested.

Issue Description
-----------------

Give a detailed description of the new feature, and what problem it is trying to solve.

If this is a proposal to change an existing feature, also outline the current
implementation and its drawbacks.

Thanks for helping us improve Mitiq! 🙂

Proposed Solution
-----------------

The `noise_scaling_converter` function in `mitiq.interfaces` is actually more general than for noise scaling functions and its name should reflect that. Also, we should consider how to make it more prominent in the documentation, as it avoids the qubit naming issues observed in `convert_from_mitiq` for Qiskit circuits. 

Additional References
---------------------

If applicable, provide some references that will help us better understand the request.

**Comment by andreamari** [comment]
Good point! Do you have name suggestions? E.g.

`in_out_converter()` ?
`make_multi_frontend()` ?

**Comment by MZuhairKhan** [comment]
Hey Misty! I would love to give this a whirl!

**Comment by andreamari** [comment]
@MZuhairKhan, it's yours!
Just to double check, did you register at https://unitaryhack.dev/ ?

**Comment by MZuhairKhan** [comment]
@andreamari Yes, I have registered this account!

**Comment by Misty-W** [comment]
@MZuhairKhan, anything we can help with?

**Comment by MZuhairKhan** [comment]
> 3\. ensuring tests cover more than just noise scaling functions.

@Misty-W, Yes, what other tests do I need to add? Will these be unit tests or something else?

**Comment by natestemen** [comment]
> what other tests do I need to add? Will these be unit tests or something else?

That's a good question, sorry it wasn't a little more clear! I think it depends on how you write the docstring and rename the function. If you put up a PR we'd be happy to provide some feedback directly on code!

**Comment by MZuhairKhan** [comment]
@natestemen Sorry for the later reply. I have just renamed the built-in test for now. You can have a look at the PR above. I'll make any changes asap, specially if you dislike the name.
